kvstore/kvstore_server.cpp

```diff
@@ -1,18 +1,32 @@
 #include <kvstore_server.h>
 
 Status KvstoreServiceImpl::put(ServerContext* context, const PutRequest* request, PutReply* reply) {
-  // TODO: put
+  std::pair<std::string, std::string> pair(request->key(), request->value());
+  std::unique_lock<std::mutex> lock(mu_);
+  map_.insert(pair);
   return Status::OK;
 }
 
 Status KvstoreServiceImpl::get(ServerContext* context, const GetRequest* request, GetReply* reply) {
-  // TODO: get
-  return Status::OK;
+  std::unique_lock<std::mutex> lock(mu_);
+  auto it = map_.find(request->key());
```

Comment:
refer to about comment about abstraction

Reply:
Fixed. i think

Reply:
I think maybe a commit wasn't pushed?  It appears to be the same as before.

Reply:
I'm confused. Before I used std::unorderedmap and it's find function (for the get function), but then i created a new kvmap class with a function get that returns an optional.